cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -2505,21 +2505,79 @@ impl MachInst for Inst {
     const TRAP_OPCODE: &'static [u8] = &[0x0f, 0x0b];
 }
 
-/// State carried between emissions of a sequence of instructions.
-#[derive(Default, Clone, Debug)]
-pub struct EmitState {
-    /// Addend to convert nominal-SP offsets to real-SP offsets at the current
-    /// program point.
-    pub(crate) virtual_sp_offset: i64,
-    /// Offset of FP from nominal-SP.
-    pub(crate) nominal_sp_to_fp: i64,
-    /// Safepoint stack map for upcoming instruction, as provided to `pre_safepoint()`.
-    stack_map: Option<StackMap>,
-    /// Current source location.
-    cur_srcloc: RelSourceLoc,
-    /// Only used during fuzz-testing. Otherwise, it is a zero-sized struct and
-    /// optimized away at compiletime. See [cranelift_control].
-    ctrl_plane: ControlPlane,
+pub use emit_state::EmitState;
+mod emit_state {
```

Comment:
I'd be inclined to move the new `emit_state` module to a separate source file rather than keeping it inline here. I don't see inline modules in Rust source very often so I had to think about why you needed a module at all (and figured out the answer). If it's a separate file then I won't wonder about it…

Reply:
Good call, done!